Empty resource bundles fail pod target to lint / build

# Report

## What did you do?

1. Define a podspec with resources bundles, whose wildcards don't match any actual file.
2. Run `pod lib lint`.

## What did you expected to happen?

One of these scenarios:
1. that the lint fails gracefully and points me to the wrong resources bundles definition in my podspec
2. that the lint and the build succeeds, because we just don't generate a resource bundle target, if it doesn't match any files, but the linter should still warn at least in that case and point me to the wrong resources bundles definition in my podspec
3. that the lint and the build succeeds, because the pod target generator adds some dummy file to the resource bundle target to trick Xcode into scaffolding the file structure for targets without any files and could be deleted then by a shell script build phase later on

## What happened instead?

A resources bundle target is created, but empty, and is not actually build by Xcode or at least doesn't generate a build artifact which can be embedded, so that the build of the pod target fails subsequently due to missing this build product. This is not shown by default when running the linter and appears hidden behind a failure of `xcodebuild`. e.g.
```
- ERROR | [OSX] xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.
```

In verbose mode:
```
CpResource /var/folders/64/vmn77sp112xdn5np6mjtwxfw0000gn/T/CocoaPods/Lint/build/Release/RxRealm.bundle /var/folders/64/vmn77sp112xdn5np6mjtwxfw0000gn/T/CocoaPods/Lint/build/Release/RxRealm.framework/Versions/A/Resources/RxRealm.bundle
    cd /var/folders/64/vmn77sp112xdn5np6mjtwxfw0000gn/T/CocoaPods/Lint/Pods
    builtin-copy -exclude .DS_Store -exclude CVS -exclude .svn -exclude .git -exclude .hg -strip-debug-symbols -strip-tool /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip -resolve-src-symlinks /var/folders/64/vmn77sp112xdn5np6mjtwxfw0000gn/T/CocoaPods/Lint/build/Release/RxRealm.bundle /var/folders/64/vmn77sp112xdn5np6mjtwxfw0000gn/T/CocoaPods/Lint/build/Release/RxRealm.framework/Versions/A/Resources
error: /var/folders/64/vmn77sp112xdn5np6mjtwxfw0000gn/T/CocoaPods/Lint/build/Release/RxRealm.bundle: No such file or directory

** BUILD FAILED **
The following build commands failed:
    CpResource /var/folders/64/vmn77sp112xdn5np6mjtwxfw0000gn/T/CocoaPods/Lint/build/Release/RxRealm.bundle /var/folders/64/vmn77sp112xdn5np6mjtwxfw0000gn/T/CocoaPods/Lint/build/Release/RxRealm.framework/Versions/A/Resources/RxRealm.bundle
```

## Notes

**2.** (and **3.**, but 😨) have the advantage that it would support podspecs, which rely on bootstrapping the contents of the resource bundle by a `prepare_command`, which could yield empty contents as a possible outcome.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,7 +8,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Enhancements
 
-* None.  
+* Validate that resource bundles declared in the podspec contain resources.  
+  [Samuel Giddins](https://github.com/segiddins)
+  [#5218](https://github.com/CocoaPods/CocoaPods/issues/5218)
 
 ##### Bug Fixes
 

--- a/lib/cocoapods/validator.rb
+++ b/lib/cocoapods/validator.rb
@@ -556,6 +556,13 @@ module Pod
       end
     end
 
+    def _validate_resource_bundles
+      file_accessor.resource_bundles.each do |bundle, resource_paths|
+        next unless resource_paths.empty?
+        error('file patterns', "The `resource_bundles` pattern for `#{bundle}` did not match any file.")
+      end
+    end
+
     # Ensures that a list of header files only contains header files.
     #
     def _validate_header_files(attr_name)

--- a/spec/unit/validator_spec.rb
+++ b/spec/unit/validator_spec.rb
@@ -642,6 +642,23 @@ module Pod
           validator.results.map(&:to_s).first.should.match /Unexpected file extension for modulemap file \(JSONKit\.m\)/
           validator.result_type.should == :error
         end
+
+        it 'checks resource bundles have resources' do
+          file = write_podspec(stub_podspec(/.*source_files.*/, <<-JSON))
+            "source_files": "JSONKit.*",
+            "resource_bundles": {
+              "bundle1": ["CHANGELOG.md", "*.md"],
+              "bundle2": "foo.bar*"
+            },
+          JSON
+
+          validator = Validator.new(file, config.sources_manager.master.map(&:url))
+          validator.stubs(:build_pod)
+          validator.stubs(:validate_url)
+          validator.validate
+          validator.results.map(&:to_s).first.should.include 'The `resource_bundles` pattern for `bundle2` did not match any file'
+          validator.result_type.should == :error
+        end
       end
 
       it 'validates a podspec with dependencies' do

